src/ui/components/Dashboard/Dashboard.jsx

```diff
@@ -1,61 +1,151 @@
-import React, { useState, useMemo, memo } from 'react';
+import React, { useState, memo, useCallback, useRef } from 'react';
+import {
+  Button,
+  FormControl,
+  Input,
+  InputLabel,
+  MenuItem,
+  Paper,
+  Select,
+  Snackbar
+} from '@material-ui/core';
+import MuiAlert from '@material-ui/lab/Alert';
 import useStyles from './styles';
-import Sidebar from './Sidebar';
-import NavBar from './NavBar';
-import Collections from '../Collections';
+import usePlanDefinitions from '../../hooks/usePlanDefinitions';
+import axios from 'axios';
+
 function Dashboard() {
   const classes = useStyles();
-  const [contentKey, setContentKey] = useState('');
-  // can do useMemo to update labels like notifications
-  const tabs = [
-    { key: 'overview', label: 'Overview', component: null },
-    { key: 'dashboard', label: 'Dashboard', component: <div>Dashboard content</div> },
-    { key: 'notifications', label: 'Notifications', component: <div>Notifications</div> },
-    { key: 'collections', label: 'Collections', component: null },
-    { key: 'servers', label: 'Servers', component: <Collections selectedCollection="servers" /> },
-    {
-      key: 'endpoints',
-      label: 'Endpoints',
-      component: <Collections selectedCollection="endpoints" />
-    },
-    {
-      key: 'subscriptions',
-      label: 'Subscriptions',
-      component: <Collections selectedCollection="subscriptions" />
-    },
-    {
-      key: 'plandefinitions',
-      label: 'Plan Definitions',
-      component: <Collections selectedCollection="plandefinitions" />
+  const importFileRef = useRef(null);
+  const [fileName, setFileName] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [alertMessage, setAlertMessage] = useState('');
+  const [alertSeverity, setAlertSeverity] = useState('');
+  const [planDefFullUrl, setPlanDefFullUrl] = useState('');
+  const { data: planDefinitions } = usePlanDefinitions();
+
+  const handleChangePlanDefFullUrl = useCallback(
+    event => {
+      setPlanDefFullUrl(event.target.value);
     },
-    { key: 'logs', label: 'Logs', component: <Collections selectedCollection="logs" /> },
-    {
-      key: 'completedreports',
-      label: 'Completed Reports',
-      component: <Collections selectedCollection="completedreports" />
+    [setPlanDefFullUrl]
+  );
+
+  const handleChooseFile = useCallback(() => {
+    if (importFileRef?.current?.files?.[0]) setFileName(importFileRef.current.files[0].name);
+  }, [importFileRef, setFileName]);
+
+  const handleTrigger = useCallback(
+    event => {
+      event.preventDefault();
```

Comment:
One thing I noticed was that you can click the trigger button without choosing a file or a PlanDef, and it just hung. We don't need to get all fancy with disabling buttons or any of that, but could there be a simple `if` check here to make sure both the plandef and patient/encounter are set before making the request, and if not post an alert saying something like "PlanDefinition and Patient/Encounter are required"?